There is a  wrong target name in shared build.

I met a compilation error on shared build.
```
CMake Error at src/CMakeLists.txt:44 (target_link_libraries):
  Cannot specify link libraries for target "chunkio-static" which is not
  built by this project
```

It seems to be caused by wrong target name on shared build.
https://github.com/fluent/chunkio/blob/v1.1.6/src/CMakeLists.txt#L44

I think it should be `target_link_libraries(chunkio-shared ${libs})`.

@edsiper 
I met the error with `CIO_LIB_STATIC=OFF` and `CIO_LIB_SHARED=ON`.

```
> cmake -DCIO_LIB_STATIC=OFF -DCIO_LIB_SHARED=ON -S . -B build
(snip)
-- The C compiler identification is GNU 13.2.1
-- The CXX compiler identification is GNU 13.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CIO_HAVE_GETPAGESIZE
-- Performing Test CIO_HAVE_GETPAGESIZE - Success
-- Performing Test CIO_HAVE_FALLOCATE
-- Performing Test CIO_HAVE_FALLOCATE - Success
CMake Error at src/CMakeLists.txt:44 (target_link_libraries):
  Cannot specify link libraries for target "chunkio-static" which is not
  built by this project.
```

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -42,7 +42,7 @@ endif()
 
 if (CIO_LIB_SHARED)
   add_library(chunkio-shared SHARED ${src})
-  target_link_libraries(chunkio-static ${libs})
+  target_link_libraries(chunkio-shared ${libs})
   if(CIO_SANITIZE_ADDRESS)
     add_sanitizers(chunkio-shared)
   endif()

--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -2,4 +2,9 @@ set(src
   cio.c)
 
 add_executable(cio ${src})
-target_link_libraries(cio chunkio-static)
+
+if (CIO_LIB_SHARED)
+  target_link_libraries(cio chunkio-shared)
+elseif (CIO_LIB_STATIC)
+  target_link_libraries(cio chunkio-static)
+endif()

